packages/protocol/contracts/stability/SortedOracles.sol

```diff
@@ -278,49 +269,13 @@ contract SortedOracles is ISortedOracles, Ownable, Initializable {
    * @dev This can be used to delete elements for oracles that have been removed.
    */
   function removeReport(address token, address oracle) private {
-    SortedFractionMedianList.Element memory originalMedian = getMedianElement(rates[token]);
+    uint256 originalMedian = rates[token].getMedianValue();
     rates[token].remove(oracle);
     timestamps[token].remove(oracle);
     emit OracleReportRemoved(token, oracle);
-    emitIfMedianUpdated(token, originalMedian);
-  }
-
-  /**
-   * @notice Emits the MedianUpdated event if the median rate has changed.
-   * @param token The address of the token for which the median rate may have changed.
-   * @param originalMedian The original median rate.
-   */
-  function emitIfMedianUpdated(
-    address token,
-    SortedFractionMedianList.Element memory originalMedian
-  )
-    private
-  {
-    SortedFractionMedianList.Element memory newMedian = getMedianElement(rates[token]);
-    if (
-      originalMedian.numerator != newMedian.numerator ||
-      originalMedian.denominator != newMedian.denominator
-    ) {
-      emit MedianUpdated(
-        token,
-        newMedian.numerator,
-        newMedian.denominator
-      );
-    }
-  }
-
-  /**
-   * @notice Returns the key for the lesser element in the timestamp list.
-   * @param token The address of the token for which the Celo Gold exchange rate is being reported.
-   * @param oracle The address of the oracle to sort into the timestamp list.
-   * @return The key of the lesser element in the list.
-   */
-  function getLesserTimestampKey(address token, address oracle) private view returns(address) {
-    address head = timestamps[token].head;
-    if (head == oracle) {
-      return timestamps[token].elements[head].lesserKey;
-    } else {
-      return head;
+    uint256 newMedian = rates[token].getMedianValue();
+    if (newMedian != originalMedian) {
+      emit MedianUpdated(token, newMedian, numRates(token) == 0 ? 0 : DENOMINATOR);
```

Comment:
Ditto here, it seems to me like behavior around empty should be embedded rather than signalled in logs in this way

Reply:
Agreed. It's invalid for SortedOracles to not return an exchange rate. Any suggestions for what to do here?

Reply:
Currently `removeExpiredReports` is the only caller of `removeReport` and requires that the number to remove is strictly less than the number of reports.